### PR TITLE
Fix crate compilation on stable, port to edition 2018

### PR DESCRIPTION
**Pull request by tmplt**
This branch fixes crate/example compilation for latest stable compiler release (1.32.0). RTFM example has been updated to use RTFM v0.4.0 (also somewhat restructured).

Crate version bumped to v0.7.0.

### CODE REVIEW

**Comment by tmplt** [comment]
An alternative to the `loop {}` that substituted the `core::intrinsics::abort()` in `src/panic.rs` could be [`extern crate panic_halt;`](https://github.com/korken89/panic-halt/blob/master/src/lib.rs).

**Comment by tmplt** [comment]
Testing `examples/serial.rs` with `bobbin console` prints everything up to `I count: 255` but not the panic which I presume should be printed also. I think I might have removed a macro somewhere.

File: Cargo.toml [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
+authors = ["Kjetil Kjeka <kjetilkjeka@gmail.com>", "Tmplt <tmplt@dragons.rocks>"]
 categories = ["embedded", "no-std", "hardware-support"]
 repository = "https://github.com/kjetilkjeka/s32k144evb.rs"
 description = "Board support crate for s32k144evb"
 keywords = ["arm", "cortex-m", "s32k144", "template"]
 license = "MIT OR Apache-2.0"
 name = "s32k144evb"
-version = "0.6.1"
+edition = "2018"
+version = "0.7.0"
```
Comment (kjetilkjeka) [inline Cargo.toml]: Please hold on upping this version, there might be a few more things going into this release.

File: Cargo.toml [hunk 2]
```diff
@@ -37,5 +36,4 @@ lto = true
 default = ["panic-over-serial"]
 itm = []
 panic-over-itm = ["itm"]
-abort-on-panic = ["cortex-m-rt/abort-on-panic"]
```
Comment (kjetilkjeka) [inline Cargo.toml:40]: Is the "abort on panic" behavior removed?

If this was necessary, please explain why and note it in the Changelog.

File: examples/can.rs [hunk 3]
```diff
@@ -1,61 +1,49 @@
...
 
-use cortex_m::asm;
+use cortex_m_rt::entry;
 
-use s32k144evb::{
-    can,
-    wdog,
-    spc,
-};
+use s32k144evb::{can, spc, wdog};
```
Comment (kjetilkjeka) [inline examples/can.rs:11]: Is this the correct format according to rustfmt?

Same on lines below.

File: examples/led.rs [hunk 4]
```diff
@@ -1,43 +1,35 @@
...
 
-use s32k144evb::{
-    led,
-    wdog,
-};
-
-use s32k144evb::pcc::Pcc;
+use cortex_m_rt::entry;
 
+use s32k144evb::{led, wdog};
```
Comment (kjetilkjeka) [inline examples/led.rs:10]: This as well.

**Review by kjetilkjeka — CHANGES_REQUESTED** [review]
Thanks for the PR.

First of all, I'm sorry for this crate being a bit mess, glad that you were able to make improvements on it nontheless :+1: 

I've left some comments. Summed up:
 - Is putting all deps on same line recommended from rustfmt? Not sure if it improves anything?
 - What happened to `abort-on-panic`?
 - Do you mind updating the changelog as well?
 - Please hold on upping the version number a little.


Other than that, I haven't got around to actually test the changes myself yet. I will probably end up running the examples on both versions and if they behave the same the test coverage is good enough for this project in its current state.

File: examples/can.rs (hunk 3 above)
Comment (tmplt) [reply examples/can.rs:11]: Yes, this was formatted via `cargo fmt`.

File: examples/led.rs (hunk 4 above)
Comment (tmplt) [reply examples/led.rs:10]: Ditto.

File: Cargo.toml (hunk 1 above)
Comment (tmplt) [reply Cargo.toml]: Reverted back to v0.6.1.

File: Cargo.toml (hunk 2 above)
Comment (tmplt) [reply Cargo.toml:40]: The `abort-on-panic` feature has been removed from `cortex-m-rt`. I believe the canonical approach is to either `extern crate panic_halt;` or to define your own (e.g., `src/panic.rs`).

I'll explain this in the changelog.

**Comment by tmplt** [comment]
> Testing `examples/serial.rs` with `bobbin console` prints everything up to `I count: 255` but not the panic which I presume should be printed also. I think I might have removed a macro somewhere.

The panic handlers must in current stable be marked with `#[panic_handler]` which expects a function with the signature `fn(&PanicInfo) -> !`. Working on that now; a panic is thrown in the svd2rust crate when stealing the peripherals.

**Comment by tmplt** [comment]
> First of all, I'm sorry for this crate being a bit mess, glad that you were able to make improvements on it nontheless

Expect more in the future! Currently working on a project that ends in March somewhere. Focus lies on CAN and the board's security module. Thanks for reviewing.

**Comment by tmplt** [comment]
Review has been addressed. Mentioned panic implementation has been updated; all examples now panic over serial by default.

**Comment by kjetilkjeka** [comment]
Sorry for being unreachable. I delivered by master thesis yesterday and moving the whole weekend. I will come back to this during the start of the upcoming week.

**Comment by kjetilkjeka** [comment]
Finally got around to test things. It worked great, and the previous version doesn't even build. Thanks a ton @Tmplt for getting this in working order!

Sorry for taking so long with this. My excuse is that I've delivered my master thesis, moved and started a new job the last two weeks. However, it's not ideal that I'm the only person that can merge these PRs, and finding a co-maintainer for `s32k144` and `s32k144evb` would be great. If you intend to use `s32k` with Rust in the future and would like to be involved on the maintainer side of these crates send me an e-mail and we will take it from there.

I will publish a new version on crates.io tonight. Just in case you have any objections and want to get another PR in first.

**Review by kjetilkjeka — APPROVED** [review]

**Comment by tmplt** [comment]
>Sorry for taking so long with this.

No worries. Thanks for taking some time out of your busy schedule!

> Just in case you have any objections and want to get another PR in first.

Nothing for now. Future PRs are likely to be feature additions.

> If you intend to use s32k with Rust in the future and would like to be involved on the maintainer side of these crates send me an e-mail and we will take it from there.

I'll be in touch!

**Pull request merged**